Frends.Community.Email.Tests/ExchangeEmailTests.cs

```diff
@@ -0,0 +1,651 @@
+﻿using Azure.Identity;
+using Microsoft.Graph;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Directory = System.IO.Directory;
+using File = System.IO.File;
+using Path = System.IO.Path;
+
+namespace Frends.Community.Email.Tests
+{
+    [TestFixture]
+    public class ExchangeEmailTests
+    {
+        private readonly string _mailbox = "inbox";
+        private readonly string _username = "frends_exchange_test_user@frends.com";
+        private readonly string _password = "y6xE58eS0TP002XrXPXhRJigI";//Environment.GetEnvironmentVariable("Exchange_User_Password");
+        private readonly string _applicationID = "3c7863bb-dc4b-4b22-98e3-52ac77235bf2";//Environment.GetEnvironmentVariable("Exchange_Application_ID");
+        private readonly string _tenantID = "97759401-0ff9-42fb-8eae-9163e29d19bf";//Environment.GetEnvironmentVariable("Exchange_Tenant_ID");
+
+        [OneTimeSetUp]
+        public void OneTimeSetup()
+        {
+            if (string.IsNullOrEmpty(_applicationID) ||
+                string.IsNullOrEmpty(_password) ||
+                string.IsNullOrEmpty(_tenantID))
+            {
+                throw new ArgumentException("Password, Application ID or Tenant ID is missing. Please check environment variables.");
+            }
+        }
+
+        [Test]
+        public async Task ReadEmailFromExchangeServer_ShouldReadOneItemTest()
+        {
+            var subject = "One Email Test";
+            await SendTestEmail(subject);
+            var settings = new ExchangeSettings
+            {
+                TenantId = _tenantID,
+                AppId = _applicationID,
+                Username = _username,
+                Password = _password,
+                Mailbox = _mailbox
+            };
+            var options = new ExchangeOptions
+            {
+                MaxEmails = 1,
+                DeleteReadEmails = false,
+                GetOnlyUnreadEmails = false,
+                MarkEmailsAsRead = false,
+                IgnoreAttachments = true,
+                EmailSubjectFilter = subject
+            };
+
+            var result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.That(result.Count, Is.EqualTo(1));
+            await DeleteMessages(subject);
+        }
+
+        [Test]
+        public async Task ReadEmailFromExchangeServer_ShouldReadFiveItemsTest()
+        {
+            var subject = "Five Emails test";
+            for (int i = 0; i < 5; i++)
```

Comment:
I think you should send more emails than just 5. Or get a smaller amount of emails, like e.g. 3.